### ISSUE TITLE
Recursive aliases outside module doesn't compile

### ISSUE THREAD

**Issue by silasb** [crystal-lang/crystal#6344]
Trying to use aliases outside a namespace doesn't compile correctly:

```crystal
alias Block = Array(ASTNode2)
alias ASTNode2 = Tuple(Block)

block = [] of ASTNode2
p ASTNode2.from([block])
```

will result in 

```
Error in test2.cr:6: undefined method 'from' for ASTNode2:Class

p ASTNode2.from([block])
```

Wrapping this inside a module will work:

```crystal
module Test2
  alias Block = Array(ASTNode2)
  alias ASTNode2 = Tuple(Block)

  def self.run
    block = [] of ASTNode2
    p ASTNode2.from([block])
  end
end

Test2.run
```

Using:

```
Crystal 0.25.1 [b782738ff] (2018-06-27)

LLVM: 4.0.0
Default target: x86_64-unknown-linux-gnu
```

Haven't read through https://github.com/crystal-lang/crystal/issues/5155 but this case might have been covered in that.

**Comment by asterite** [comment]
It looks like it's a recursive alias. Recursive aliases behave strange, they can't have class methods. You are lucky it works in some cases.

Recursive aliases will be eventually removed from the language, so it's better if you don't use them.

**Comment by silasb** [comment]
Yeah, it's a bummer that I made it this far without any issues with recursive aliases. If Crystal is going to eventually remove alias then the documentation shouldn't suggest using an alias for a recursive type [alias](https://crystal-lang.org/docs/syntax_and_semantics/alias.html). :(

**Comment by RX14** [comment]
How difficult would it be to fix this bug?

**Comment by asterite** [comment]
I just tried to understand when and how recursive aliases are solved/determined and I can't understand it. So for me the difficulty is "impossible" right now. But if someone else wants go give it a try, please go ahead.

**Comment by RX14** [comment]
Lets remove them.

**Comment by Nephos** [comment]
Is that going to affect `alias A = A | Array(A)` ?

**Comment by RX14** [comment]
That's a recursive alias, so yes. However, thats also a non-terminating recursive alias so I can't see how it can be used currently.
